### PR TITLE
Transform JSX spread children

### PR DESCRIPTION
**Pull request by jablko**
#11591 added support for [parsing](https://github.com/microsoft/TypeScript/pull/11591/files#diff-12a6724be007eb1a19d80018c7a63bbc73525ca793a9b3e5da49a4e86bbf457c) and [emitting](https://github.com/microsoft/TypeScript/pull/11591/files#diff-537414476d7c7a690310c5e1e5d6a0bbbfba37c5b05126edee32fdea7e3a09d8) [JSX spread children](https://github.com/facebook/jsx/blob/master/AST.md#:~:text=a%20jsx%20element%20uses%20a%20special%20form%20of%20an%20expression%20container%20for%20an%20iterator%20child%20who%20should%20be%20%E2%80%9Cspread%20out%E2%80%9D%20inside%20an%20element%E2%80%99s%20children%20list%3A), however `transformers/jsx.ts` ignores `JsxExpression#dotDotDotToken`, so roughly it transforms `<div>{...todos}</div>` -> `React.createElement("div", null, todos)` vs. `React.createElement("div", null, ...todos)`.

This isn't a problem for `React.createElement()` because that runtime treats both cases equivalently (it flattens children), however it is an issue for pickier runtimes.

This PR adds `factory.createSpreadElement()` to `transformers/jsx.ts` if `JsxExpression#dotDotDotToken` is defined.

### CODE REVIEW

**Review by weswigham — CHANGES_REQUESTED** [review]
A new test that doesn't downlevel all the way to the `__spreadArray` helper (so targeting es6) would be nice, just to capture the difference simply. Other than that, this looks good.

**Comment by jablko** [comment]
@weswigham Thanks! Assuming that downleveling and JSX are independent, can we add `@target: es6` to the existing test, to clarify the output? @sandersn mentioned [here](https://github.com/microsoft/TypeScript/pull/11591#issuecomment-253572493) that `tsxSpreadChildrenInvalidType.tsx` covers the `jsx: react` emit.

That's what I've done in the added commit. Otherwise I'd copy `tests/cases/conformance/jsx/tsxSpreadChildren.tsx` -> `tsxSpreadChildren_es6.tsx` and add:
```typescript
// @jsx: react
// @target: es6
```

Not sure if retaining both targets (default and es6) adds anything though?

File: tests/cases/conformance/jsx/tsxSpreadChildrenInvalidType.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // @jsx: react
+// @target: es6
```
Comment (rbuckton) [inline tests/cases/conformance/jsx/tsxSpreadChildrenInvalidType.tsx:2]:
```suggestion
// @target: es2015,es5
```
Targeting both doesn't hurt and gives us a baseline for any future quirky regressions.

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: tests/cases/conformance/jsx/tsxSpreadChildrenInvalidType.tsx (hunk 1 above)
Comment (jablko) [reply tests/cases/conformance/jsx/tsxSpreadChildrenInvalidType.tsx]: Thanks! Done. :heavy_check_mark:

**Review by weswigham — APPROVED** [review]

**Pull request merged**